sknn/tests/test_data.py

```diff
@@ -60,8 +61,8 @@ def test_SetLayerParamsList(self):
         nn.set_parameters([(weights, biases)])
         
         p = nn.get_parameters()
-        assert_true((p[0].weights == weights).all())
-        assert_true((p[0].biases == biases).all())
+        assert_true((p[0].weights == weights.astype(theano.config.floatX)).all())
```

Comment:
Oh, that's a real problem. If users forcibly change the data type of input `weights` and 'biases' to a mismatched version with `get_parameters()` returned value, the `assert_true()` function would be false again.